Lesson06/Lesson06/Lesson06/BankAccount.cs

```diff
@@ -0,0 +1,162 @@
+﻿
+/*
+ЗАДАНИЕ:
+Для класса банковский счет переопределить операторы == и != для
+сравнения информации в двух счетах. Переопределить метод Equals
+аналогично оператору ==, не забыть переопределить метод
+GetHashCode(). Переопределить метод ToString() для печати
+информации о счете. Протестировать функционирование
+переопределенных методов и операторов на простом примере.
+*/
+
+/*
+КОММЕНТАРИЙ-ВОПРОС:
+Я вначале переопределил Equals кодом указанным в самом низу, по примеру из методички ООП урока номер 6 на 21(22) странице.
+Там написано "В данном случае это именно перегрузка, потому что ни один из вариантов метода Equals () не принимал 
+объект класса Car." При использовании слова "override" компилятор ругался, потому сделал как в методичке, и всё работало.
+Тем более в задании написано "Переопределить метод Equals аналогично оператору ==".
+Компилятор даже говорил, что есть перегрузка метода.
+Но потом я заметил что он подчёркивает зелёным само название моего класа и пишет что мой класс "определяет
+операторы "==" и "!=", но не определяет Object.Equals(object o)". Я в итоге разобрался, и сделал как просит компилятор.
+А в методичке ошибка получается? Или всё же оба способа имеют право на существование?
+*/
+
+namespace Lesson06
+{
+
+    public class BankAccount
+    {
+
+        private int _numberOfAccount;//номер счёта
+        private decimal _balanceOfAccount;//баланс
+        public enum AccountType//enum не сделать приватным, иначе не работает программа
+        {
+            Debit, Credit, Brokerage, Crypt
+        }
+        private AccountType _accType;//потому для типа счёта создаём поле/переменную на базе созданного enum
+                                     //и уже она становится приватной
+
+        public int NumberOfAccount
+        {
+            get { return _numberOfAccount; }
+            set { _numberOfAccount = value; }
+        }
+
+        public decimal BalanceOfAccount
+        {
+            get { return _balanceOfAccount; }
+            set { _balanceOfAccount = value; }
+        }
+
+        public AccountType AccType
+        {
+            get { return _accType; }
+            set { _accType = value; }
+
+        }
+        /// <summary>
+        /// Сравнение счетов соответственно по значениям всех полей.
+        /// </summary>
+        /// <param name="Number1">Первый счёт</param>
+        /// <param name="Number2">Второй счёт</param>
+        /// <returns>Если значения всех полей соответственно равны - true, если нет - false.</returns>
+        public static bool operator ==(BankAccount Number1, BankAccount Number2)
+        {
+            if ((Number1.AccType == Number2.AccType)
+                &&
+                (Number1.BalanceOfAccount == Number2.BalanceOfAccount)
+                 &&
+                (Number1.NumberOfAccount == Number2.NumberOfAccount))
+            {
+                return true;
+            }
+            else
+            { return false; }
+
+        }
```

Comment:
Индусский код. В круглых скобках оператора `if` уже находится `bool`-значение, почему бы его просто не вернуть?
```suggestion
        public static bool operator ==(BankAccount Number1, BankAccount Number2)
        {
            return Number1.AccType == Number2.AccType
                && Number1.BalanceOfAccount == Number2.BalanceOfAccount
                 && Number1.NumberOfAccount == Number2.NumberOfAccount;
        }
```

Reply:
Но при этом если в пером, или во втором операнде у Вас будет `null`, то это обрушит всё приложение.
Данный оператор обычно реализует вызов метода `Equals`, а вся логика сравнения описывается в нём.
```suggestion
        public static bool operator ==(BankAccount Number1, BankAccount Number2)
        {
            return Equals(Number1, Number2);
        }
```